src/components/post-include-controls.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * WordPress dependencies
+ */
+import { FormTokenField, BaseControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Generates a post include control component.
+ *
+ *@return {Element} PostIncludeControls
+ */
+
+export const PostIncludeControls = ( { attributes, setAttributes } ) => {
+	const {
+		query: {
+			include_posts: includePosts = [],
+			postType,
+			multiple_posts: multiplePosts = [],
+			exclude_current: excludeCurrent = 0,
+		} = {},
+	} = attributes;
+
+	const posts = useSelect(
+		( select ) => {
+			const { getEntityRecords } = select( 'core' );
+
+			return [ ...multiplePosts, postType ].reduce(
+				( totalRecords, currentPostType ) => {
+					const records = getEntityRecords(
+						'postType',
+						currentPostType,
+						{
+							per_page: -1,
```

Comment:
This will be dangerous on sites with large datasets. Rather than load all of the posts for all of the selected post types, perhaps we should start with 10 most recent and allow the user to search as well?